src/modules/hyprland/workspaces.cpp

```diff
@@ -303,34 +310,48 @@ void Workspaces::onWorkspaceDestroyed(std::string const &payload) {
 
 void Workspaces::onWorkspaceCreated(std::string const &workspaceName,
                                     Json::Value const &clientsData) {
-  const Json::Value workspacesJson = gIPC->getSocket1JsonReply("workspaces");
+  spdlog::debug("Workspace created: {}", workspaceName);
+  auto const workspacesJson = gIPC->getSocket1JsonReply("workspaces");
 
   if (!isWorkspaceIgnored(workspaceName)) {
+    auto const workspaceRules = gIPC->getSocket1JsonReply("workspacerules");
     for (Json::Value workspaceJson : workspacesJson) {
       std::string name = workspaceJson["name"].asString();
       if (name == workspaceName &&
           (allOutputs() || m_bar.output->name == workspaceJson["monitor"].asString()) &&
           (showSpecial() || !name.starts_with("special")) && !isDoubleSpecial(workspaceName)) {
+        for (Json::Value const &rule : workspaceRules) {
+          if (rule["workspaceString"].asString() == workspaceName) {
+            workspaceJson["persistent"] = rule["persistent"].asBool();
+            break;
+          }
+        }
+
         m_workspacesToCreate.emplace_back(workspaceJson, clientsData);
         break;
       }
     }
+  } else {
+    spdlog::debug("Not creating workspace because it is ignored: {}", workspaceName);
   }
```

Comment:
To facilitate, this is the code block I was referring to when I mentioned that it isn't in master, but that I firmly believe should include the extendOrphans method